engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/HashMapInsertAllNode.java

```diff
@@ -0,0 +1,73 @@
+package org.enso.interpreter.runtime.data.hash;
+
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import org.enso.common.LanguageInfo;
+import org.enso.interpreter.node.expression.builtin.meta.EqualsNode;
+import org.enso.interpreter.node.expression.builtin.meta.HashCodeNode;
+
+@GenerateUncached
+@NodeInfo(
+    shortName = "MapInsertAll",
+    description = "Inserts all elements from the given container into a hash map",
+    language = LanguageInfo.ID)
+public abstract class HashMapInsertAllNode extends Node {
+
+  public static HashMapInsertAllNode build() {
+    return HashMapInsertAllNodeGen.create();
+  }
+
+  public static HashMapInsertAllNode getUncached() {
+    return HashMapInsertAllNodeGen.getUncached();
+  }
+
+  /**
+   * Insert all the elements from the given container into the given map.
+   *
+   * @param self A map to insert into.
+   * @param container Either a map, or a list of pairs to insert into the map.
+   * @param maxItems Maximum number of items to insert into the map from the container.
+   */
+  public abstract EnsoHashMap executeInsertAll(
+      VirtualFrame frame, Object self, Object container, int maxItems);
+
+  @Specialization
+  EnsoHashMap doEnsoHashMaps(
```

Comment:
Only a single specialization and no fallback? Have you tried to change signature of `executeInsertAll` to accept only `EnsoHashMap`s?

Reply:
Good observation. Signature changed in ac6dfb42c6c7f983dfac0513927c4841d115ddcd